### ISSUE TITLE
Add support for the Truffle development environment

### ISSUE THREAD

**Issue by olawiberg** [the-chaingang/ethereal#31]
Truffle can connect to any given network node when publishing contracts using the migration process. This node needs to meet some minimum requirements to work. Some conditions that have to be met are:

At least one account needs to exists.
That account must be unlocked on the node that Truffle connects to.
There need to be sufficient funds on that account to perform the calculations needed to deploy the contract and perform other calculations.

There may be more requirements needed.

**Comment by zomglings** [comment]
@olawiberg : Can we turn this into an outline for a test case? Something like this:

1. Bring up an `ethereal` network with 2 nodes.

1. Seed the network with some specified accounts. (Do we need a mining node or can we add money to the appropriate accounts in the genesis file?)

1. Import a particular smart contract that we can enact between accounts.

1. Execute the smart contract.

1. How do we test the results?

We should ideally have something even more concrete than the steps above.

**Comment by olawiberg** [comment]
@nkashy1 Yes, it is a great idea to do it as a test case. In order to enable development, there needs to be some at least one account with a positive balance of Ether in order to deploy the contracts and perform transactions. Your suggestion looks good as a starting outline. It would be a good idea to start a mining node and make the mining api available so that the coinbase account can be used for testing without seeding with other accounts. I would suggest the following steps:

1. Bring up an ethereal network with 2 nodes, one being a mining node with the miner/admin account. The mining node will need a coinbase account.
1. Seed the network with two additional accounts, one which should have positive balance.
1. Import a particular smart contract that we can enact between accounts.
1. Execute the smart contract that transfer funds from one account to another.
1. Test the results by verifying the balance in the accounts before and after the transaction and verify the information in the transaction block

**Comment by zomglings** [comment]
In order to really be able to test that truffle is working properly, we are going to have to make sure that:

1. We can specify a `genesis.json` file which seeds some predetermined accounts with coins (this will involve writing a test with a custom genesis file in which the appropriate accounts have the the appropriate balance).

1. Perhaps be able to trigger a signal once mining has started?

Besides that, we can do the truffle portion of the verification manually for now. How does that sound?

**Comment by zomglings** [comment]
@olawiberg : On second thought, it's not clear to me that we need to use a mining node to verify that smart contracts execute correctly. Do we need the mining node?

**Comment by olawiberg** [comment]
@nkashy1 the mining node is not necessary. There are two option, either we make it easy to add accounts with balances in the `genesis.json` file, or use the mining node to mine Ether that can be transferred to other accounts. The most lightweight solution is to use preconfigured accounts so let's start with that setup.

**Comment by zomglings** [comment]
Agreed, let me create an issue for it.

**Comment by zomglings** [comment]
@olawiberg : #32 , please take a look
